components/raftstore-v2/src/operation/ready/snapshot.rs

```diff
@@ -381,8 +396,20 @@ impl<EK: KvEngine, ER: RaftEngine> Storage<EK, ER> {
         self.entry_storage_mut().set_truncated_term(last_term);
         self.entry_storage_mut().set_last_term(last_term);
 
-        let key = TabletSnapKey::new(region_id, peer_id, last_term, last_index);
-        let mut path = snap_mgr.final_recv_path(&key);
+        let (path, clean_split) = match self.split_init_mut() {
```

Comment:
Q: is it possible both apply snapshot and split is on-going? Then here we use split first. The snapshot will be ignored and the disk space could be leaked forever.

Reply:
Yes it's possible. The snapshot is not leaked here, but the very moment in `step`. The cleanup should be done there.

Reply:
Where exactly? I thought currently the tablet created in `recv_snap` is assumed to be always loaded.

Reply:
https://github.com/tikv/tikv/blob/280d53b75c1d3e821435c440eb6493af536edabb/components/raftstore-v2/src/operation/ready/mod.rs#L66-L128

There is no such guarantee, a snapshot can be discarded for various reasons, conflict with split is just one of them.

Reply:
I mean where is it physically deleted? If it should be done by tablet GC then why clean up the split tablet here during apply?

Reply:
It's not deleted yet. It's one of the known leak problem and should be fixed in another PR.